vizro-core/src/vizro/models/_dashboard.py

```diff
@@ -114,14 +118,41 @@ def build(self):
             fluid=True,
         )
 
+    def _make_page_layout(self, page):
+        # Identical across pages
+        dashboard_title = (
+            html.Div(children=[html.H2(self.title), html.Hr()], className="dashboard_title", id="dashboard_title_outer")
+            if self.title
+            else html.Div(className="hidden", id="dashboard_title_outer")
+        )
+        theme_switch = daq.BooleanSwitch(
+            id="theme_selector", on=True if self.theme == "vizro_dark" else False, persistence=True
+        )
+
+        # Shared across pages but slightly differ in content
+        page_title = html.H2(children=page.title, id="page_title")
+        navigation = cast(Navigation, self.navigation).build(active_page_id=page.id)
+
+        # Different across pages
+        page_content = page.build()
+        control_panel = page_content["control_panel_outer"]
+        component_container = page_content["component_container_outer"]
```

Comment:
I think this is a very good solution 👍 It keeps the rule that `build` returns one Dash component and is indexable like a dictionary so easy to split apart.

Please could you just double check that in Dash/React there's no component other than `Div` that's specifically designed to be a "fake container" just to group things together. I don't think there is but just to be sure that `Div` is the best thing to use for this.

Note this "return one component" rule is not just our rule but [also in Dash/React](https://dash.plotly.com/react-for-python-developers):
> Notice that we're not allowed to return multiple elements from render(), but an element with children is totally fine.
They seem to use `div` for this purpose in their example also, but maybe there's more up to date React guidance somewhere that does is slightly differently.

Reply:
Interesting point! I've just checked and using html.Divs to group things is correct and commonly used: https://www.shecodes.io/athena/1967-why-should-we-wrap-react-components-in-a-div

Right below that example where they argue for `html.Divs` though, they mention `react.fragments `which seem to work very similar to `html.Divs` but have the added advantage that they do not add any new element to the DOM.

Just looked through the dcc and html library and there don't seem to be a react.fragment element that we could use. Found this issue that has been closed: https://github.com/plotly/dash-html-components/issues/125 but it doesn't seem like they have added it into their library for users to use. 

So I think currently html.Divs should work fine 👍

Reply:
Ah yes, a fragment is just the sort of thing I was thinking of a would have been perfect for this sort of thing! Pity it never made it into `dcc`. Thanks for checking.

In this case there's no extra element cluttering the DOM because we just take the children out of the `html.Div` straight away so not a big deal. In other cases it would have been nice to have a `dcc.Fragment` to avoid the extra element but good to know that `html.Div` is still the best way to do it 👍